### PR TITLE
Use k8s.io/utils/clock in placement. Fix flaking tests

### PR DESCRIPTION
**Pull request by JoshVanL**
Same as https://github.com/dapr/dapr/pull/6059 but for `pkg/placement` and `pkg/placement/raft`.

### CODE REVIEW

**Comment by ItalyPaleAle** [comment]
@JoshVanL is this ready for review? I see it’s still a draft.

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

File: pkg/placement/placement_test.go [hunk 1]
```diff
@@ -19,26 +19,29 @@ import (
 	"testing"
 	"time"
 
+	"github.com/gogo/status"
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply pkg/placement/placement_test.go]: Thanks :+1:

File: pkg/placement/ha_test.go [hunk 2]
```diff
@@ -210,81 +240,111 @@ func TestPlacementHA(t *testing.T) {
...
 		}
 	}
-
-	return srv, ready, stopFn
 }
 
 func findLeader(t *testing.T, raftServers []*raft.Server) int {
-	n := atomic.Int32{}
-	n.Store(-1)
+	t.Helper()
```
Comment (ItalyPaleAle) [inline pkg/placement/ha_test.go:309]: I recommend not marking functions as test helpers if they contain assertions inside.

When a function is a helper, a failed test inside will be reported in the line of the caller. See docs for Helper(): https://pkg.go.dev/testing#T.Helper


```suggestion
```

File: pkg/placement/ha_test.go [hunk 3]
```diff
@@ -210,81 +240,111 @@ func TestPlacementHA(t *testing.T) {
...
 
-	return int(n.Load())
+		return true
+	}, time.Second*10, time.Second, "no leader elected")
+	return n
 }
 
 func retrieveValidState(t *testing.T, srv *raft.Server, expect *raft.DaprHostMember) {
-	actual := atomic.Pointer[raft.DaprHostMember]{}
+	t.Helper()
```
Comment (ItalyPaleAle) [inline pkg/placement/ha_test.go:338]:
```suggestion
```

File: pkg/placement/ha_test.go [hunk 4]
```diff
@@ -210,81 +240,111 @@ func TestPlacementHA(t *testing.T) {
 }
 
 func createRaftServer(t *testing.T, nodeID int, peers []raft.PeerInfo) (*raft.Server, <-chan struct{}, context.CancelFunc) {
-	srv := raft.New(fmt.Sprintf("mynode-%d", nodeID), true, peers, "")
+	t.Helper()
```
Comment (ItalyPaleAle) [inline pkg/placement/ha_test.go:243]:
```suggestion
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Some minor comments but otherwise LGTM

File: pkg/placement/ha_test.go (hunk 2 above)
Comment (JoshVanL) [reply pkg/placement/ha_test.go:309]: :+1:

File: pkg/placement/placement.go [hunk 5]
```diff
@@ -113,57 +110,75 @@ type Service struct {
...
-	if err != nil {
-		return fmt.Errorf("failed to listen: %w", err)
+// Blocks until the service is closed and all connections are drained.
+func (p *Service) Run(ctx context.Context, port string) error {
+	if !p.running.CompareAndSwap(false, true) {
+		return errors.New("placement service is already running")
 	}
 
-	opts, err := daprCredentials.GetServerOptions(certChain)
+	if p.closed.Load() {
```
[the remark this answers is not in the document]
Comment (JoshVanL) [reply pkg/placement/placement.go:154]: :+1:

**Comment by JoshVanL** [comment]
Addressed your comments, thanks @ItalyPaleAle

**Review by ItalyPaleAle — APPROVED** [review]

**Pull request merged**